tech_reports/prog_examples/sfpu_eltwise_chain/sfpu_eltwise_chain.md

```diff
@@ -0,0 +1,282 @@
+for (uint32_t i = 0; i < tt::constants::TILE_HW; i++) {
+    ptr[i] = float_to_bfloat16(1.0f);
+}
+cb_push_back(ones_cb_index, one_tile);
+```
+
+Nota additional `float_to_bfloat16` function. Since `bfloat16` is not native C++ type we obtain `uint16_t` pointer (the same size as `bfloat16`) and save converted values.
+
+---
+
+### Writer Kernel
+
+The writer kernel reads the computed result from the result circular buffer and writes it back to DRAM:
+
+```cpp
+cb_wait_front(result_cb_index, one_tile);
+const uint32_t l1_read_addr = get_read_ptr(result_cb_index);
+noc_async_write_tile(0, interleaved_accessor, l1_read_addr);
+noc_async_write_barrier();
+cb_pop_front(result_cb_index, one_tile);
+```
+
+---
+
+### Compute Kernel - The SFPU Fusion Core
+
+This is the heart of the example, demonstrating **SFPU operation chaining**:
+
+```cpp
+// Initialize SFPU and acquire tile registers
+init_sfpu(src_cb_index, result_cb_index);
+tile_regs_acquire();
+
+// Load data into registers
+cb_wait_front(src_cb_index, one_tile);
+cb_wait_front(ones_cb_index, one_tile);
+copy_tile(src_cb_index, 0, 0);      // Input data → register 0
+copy_tile(ones_cb_index, 0, 1);     // Ones tile → register 1
+
+// Fused SFPU operations chain
+exp_tile_init();
+exp_tile(0);                        // register 0 = exp(input)
+
+add_binary_tile_init();
+add_binary_tile(0, 1, 0);          // register 0 = exp(input) + 1
+
+log_tile_init();
+log_tile(0);                       // register 0 = log(exp(input) + 1)
+
+// Store result and cleanup
+tile_regs_commit();
+tile_regs_wait();
+cb_reserve_back(result_cb_index, one_tile);
+pack_tile(0, result_cb_index);
+```
+
+### Key SFPU Fusion Concepts
+
+1. **Register Reuse**: The output of each operation stays in the same register and becomes input for the next operation
+2. **No Intermediate Memory**: Results don't need to be stored back to circular buffers between operations
+3. **Operation Initialization**: Each SFPU operation requires initialization (`*_init()`) before use
+4. **Efficient Chaining**: Multiple complex functions can be computed in a single pass through the data
+
+---
+
+## Expected Output
+
+```console
+Metalium vs Golden -- PCC = 0.999847
+```
+
+The high Pearson Correlation Coefficient (>0.999) indicates that the SFPU fused operations produce results nearly identical to the CPU golden reference, demonstrating both the correctness and precision of the hardware implementation.
+
+---
+
+## Benefits of SFPU Fusion
+
+1. **Memory Bandwidth Reduction**: Eliminates intermediate memory transfers
+2. **Lower Latency**: Reduces the number of memory access cycles
+3. **Higher Throughput**: Enables computation of complex functions in fewer kernel invocations
+4. **Power Efficiency**: Reduces memory I/O operations which are typically power-intensive
+
+This pattern can be extended to implement more complex mathematical functions by chaining additional SFPU operations together.
```

Comment:
We should add a note here that Dst can either hold BFP16 or FP32 depend on settings. Precision loss is a real concern.

This doc should serve the purpose https://docs.tenstorrent.com/tt-metal/latest/tt-metalium/tt_metal/advanced_topics/compute_engines_and_dataflow_within_tensix.html#dst-register

I am working on a proper precision guide.